integration/raft/migration_test.go

```diff
@@ -69,6 +74,127 @@ var _ = Describe("ConsensusTypeMigration", func() {
+
+			By("Config update on standard channel, State=MAINTENANCE, enter maintenance-mode")
+			config, updatedConfig := prepareTransition(network, peer, orderer, "testchannel",
+				"etcdraft", protosorderer.ConsensusType_STATE_NORMAL,
+				"etcdraft", nil, protosorderer.ConsensusType_STATE_MAINTENANCE)
+			nwo.UpdateOrdererConfig(network, orderer, "testchannel", config, updatedConfig, peer, orderer)
+
+			bftMetadata := protoutil.MarshalOrPanic(&smartbft.Options{
+				RequestBatchMaxCount:      types.DefaultConfig.RequestBatchMaxCount,
+				RequestBatchMaxBytes:      types.DefaultConfig.RequestBatchMaxBytes,
+				RequestBatchMaxInterval:   types.DefaultConfig.RequestBatchMaxInterval.String(),
+				IncomingMessageBufferSize: types.DefaultConfig.IncomingMessageBufferSize,
+				RequestPoolSize:           types.DefaultConfig.RequestPoolSize,
+				RequestForwardTimeout:     types.DefaultConfig.RequestForwardTimeout.String(),
+				RequestComplainTimeout:    types.DefaultConfig.RequestComplainTimeout.String(),
+				RequestAutoRemoveTimeout:  types.DefaultConfig.RequestAutoRemoveTimeout.String(),
+				ViewChangeResendInterval:  types.DefaultConfig.ViewChangeResendInterval.String(),
+				ViewChangeTimeout:         types.DefaultConfig.ViewChangeTimeout.String(),
+				LeaderHeartbeatTimeout:    types.DefaultConfig.LeaderHeartbeatTimeout.String(),
+				LeaderHeartbeatCount:      types.DefaultConfig.LeaderHeartbeatCount,
+				CollectTimeout:            types.DefaultConfig.CollectTimeout.String(),
+				SyncOnStart:               types.DefaultConfig.SyncOnStart,
+				SpeedUpViewChange:         types.DefaultConfig.SpeedUpViewChange,
+			})
+			config, updatedConfig = prepareTransition(network, peer, orderer, "testchannel",
+				"etcdraft", protosorderer.ConsensusType_STATE_MAINTENANCE,
+				"BFT", bftMetadata, protosorderer.ConsensusType_STATE_MAINTENANCE)
+			nwo.UpdateOrdererConfig(network, orderer, "testchannel", config, updatedConfig, peer, orderer)
+
+			time.Sleep(time.Second) // TODO: check block was committed in all orderers
+
+			for _, oProc := range []ifrit.Process{o1Proc, o2Proc, o3Proc, o4Proc} {
+				if oProc != nil {
+					oProc.Signal(syscall.SIGTERM)
+					Eventually(oProc.Wait(), network.EventuallyTimeout).Should(Receive())
+				}
+			}
+
+			runOrderers()
+
+			// TODO:
+			By("Waiting for followers to see the leader, again")
+			Eventually(o2Runner.Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1 channel=testchannel"))
+			Eventually(o3Runner.Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1 channel=testchannel"))
+			Eventually(o4Runner.Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1 channel=testchannel"))
+
+			// Exit maintenance mode
+
+			config, updatedConfig = prepareTransition(network, peer, orderer, "testchannel",
+				"BFT", protosorderer.ConsensusType_STATE_MAINTENANCE,
+				"BFT", nil, protosorderer.ConsensusType_STATE_NORMAL)
```

Comment:
what happens when we set the metadata here to nil? Is it disregarded because of the change in maintenance status? @tock-ibm